external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -1707,6 +1708,21 @@ fn exp(context: &mut Context, e: Box<E::Exp>) -> Box<N::Exp> {
                             (bloc, help_msg),
                         ));
                     }
+                    // If no abort code is given for the assert, we add in the abort code as the
+                    // bitset-line-number if `CleverAssertions` is set.
+                    if nes.value.len() == 1
+                        && is_macro.is_some()
+                        && context.env.check_feature(
+                            context.current_package,
+                            FeatureGate::CleverAssertions,
+                            bloc,
+                        )
+                    {
+                        let line_no = context.env.file_mapping().location(bloc).start.line;
+                        let bitset = ErrorBitset::new(line_no as u16, u16::MAX, u16::MAX);
+                        let val = NE::Value(sp(bloc, E::Value_::U64(bitset.bits)));
+                        nes.value.push(sp(bloc, val));
```

Comment:
We can move this into typing probably (but will make a couple things a bit more complicated is all). In terms of the warnings though, couldn't that be done here in naming? Since all we are looking for are raw literals in assertions we should be able to do it anywhere was my thinking, but maybe I'm misunderstanding something.